src/demos.py

```diff
@@ -31,224 +31,173 @@
-            )
-            force_execution = choice == "y" or choice == "Y"
-            steps.append(GPTReviewSentimentAnalyzer(force_refresh=force_execution))
-    except ValueError:
+# db_demo
+def db_demo():
+    amt_leads = get_database().get_cardinality()
+    lead_id = get_int_input(f"Choose a lead_id in range [1, {amt_leads}]\n")
+    if 1 <= lead_id <= amt_leads:
+        print(get_database().get_lead_by_id(lead_id))
+    else:
         print("Invalid Choice")
 
-    try:
-        choice = str(input(f"Use the Regionalatlas? (y/N)\n"))
-        if choice == "y" or choice == "Y":
-            steps.append(RegionalAtlas(force_refresh=True))
-    except ValueError:
-        print("Invalid Choice")
 
-    limit = None
+def add_step_if_requested(
+    steps, step_classes, step_desc, step_warning_message: str = ""
+):
+    if get_yes_no_input(f"Run {step_desc} {step_warning_message}(y/N)?\n"):
+        force = get_yes_no_input("Force execution if data is present? (y/N)\n")
+        for cls in step_classes:
+            steps.append(cls(force_refresh=force))
 
-    try:
-        choice = int(input(f"Set limit for data point to be processed\n"))
-        if choice > 0:
-            limit = choice
-        else:
-            print("Invalid Choice, no limit set")
-    except ValueError:
-        print("Invalid Choice, no limit set")
 
-    if limit is None:
-        try:
-            choice = str(
-                input(
-                    f"Do you want to save the output data to S3? Will replace the current file!\n"
-                )
-            )
-            if choice == "y" or choice == "Y":
-                output_location_remote = f"s3://{S3_BUCKET}/leads/enriched.csv"
-        except ValueError:
-            print("Invalid Choice")
+# pipeline_demo
+def pipeline_demo():
+    steps = [AnalyzeEmails(force_refresh=True)]
+    additional_steps = [
+        ([ScrapeAddress], "Scrape Address", "(will take a long time)"),
+        ([FacebookGraphAPI], "Facebook Graph API", "(will use token)"),
+        ([PreprocessPhonenumbers], "Phone Number Validation", ""),
+        (
+            [GooglePlaces, GooglePlacesDetailed],
+            "Google API",
+            "(will use token and generate cost!)",
+        ),
+        (
+            [GPTReviewSentimentAnalyzer],
+            "open API Sentiment Analyzer",
+            "(will use token and generate cost!)",
+        ),
+        ([RegionalAtlas], "Regionalatlas", ""),
+    ]
+
+    if get_yes_no_input(f"Run Demo pipeline with all steps (y/N)?\n"):
+        for step_classes, _, _ in additional_steps:
+            step_instances = [cls(force_refresh=True) for cls in step_classes]
+            steps.extend(step_instances)
+    else:
+        for step_classes, desc, warning_message in additional_steps:
+            add_step_if_requested(steps, step_classes, desc, warning_message)
+
+    limit = get_int_input("Set limit for data points to be processed\n")
```

Comment:
Integer 0 has to be interpreted as no limit
```suggestion
    limit = get_int_input("Set limit for data points to be processed (0=No limit)\n")
    limit = limit if limit > 0 else None
```